serving/serving.go

```diff
@@ -153,10 +153,14 @@ func (serv *FeatureServer) getTrainingSetIterator(name, variant string) (provide
 func (serv *FeatureServer) getSourceDataIterator(name, variant string, limit int64) (provider.GenericTableIterator, error) {
 	ctx := context.TODO()
 	serv.Logger.Infow("Getting Source Variant Iterator", "name", name, "variant", variant)
-	sv, err := serv.Metadata.GetSourceVariant(ctx, metadata.NameVariant{name, variant})
+	sv, err := serv.Metadata.GetSourceVariant(ctx, metadata.NameVariant{Name: name, Variant: variant})
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get source variant")
 	}
+	// TODO: Determine if we want to add a backoff here to wait for the source
+	if sv.Status() != metadata.READY {
+		return nil, fmt.Errorf("source variant is not ready")
+	}
```

Comment:
@sdreyer, we should talk about how we want to handle the resource status. I added this here and changed code in `client.py` to ensure I wasn't trying to fetch a resource before it was ready, which is especially important in the case of transformation sources.

For your consideration, here is the code I changed in `client.py`:

```py
def dataframe(
    self,
    source: Union[SourceRegistrar, LocalSource, SubscriptableTransformation, str],
    variant: Union[str, None] = None,
    limit=NO_RECORD_LIMIT,
    synchronous=True,
):
    # ... method docstring ...
    self.apply(synchronous=synchronous)
    # ... rest of method logic ...
    return self.impl._get_source_as_df(name, variant, limit)
```

Reply:
You'll note that I changed the param name in `apply` to synchronous because it was somewhat confusing based on the existing naming and documentation. We can chat more about whether we want to include this change as is, but we'll need to handle the resource state somewhere.